src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -614,14 +615,24 @@ private static void parseSingleAltAllele(List<Allele> alleles, String alt, int l
     }
 
     public final static boolean canDecodeFile(final String potentialInput, final String MAGIC_HEADER_LINE) {
+    	FileInputStream fis = null;
+    	GZIPInputStream giz = null;
+    	BlockCompressedInputStream bcis = null;
         try {
-            return isVCFStream(new FileInputStream(potentialInput), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new GZIPInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new BlockCompressedInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE);
+        	fis = new FileInputStream(potentialInput);
```

Comment:
I don't see why not make them now that you're in the codebase. You've already fixed a whole bunch of resource leaks in this branch, which is great.